### PR TITLE
Refactor the functions for future changes

### PR DESCRIPTION
**Pull request by raghavendra-talur**
These are some of the changes that we need for the Metro DR patches. Creating a separate PR for existing code to make the review easier.

### CODE REVIEW

File: controllers/drpolicy_controller.go [hunk 1]
```diff
@@ -77,7 +77,8 @@ func (r *DRPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			return ctrl.Result{}, fmt.Errorf("finalizer add update: %w", u.validatedSetFalse("FinalizerAddFailed", err))
 		}
 
-		if reason, err := validate(ctx, drpolicy, r.APIReader, r.ObjectStoreGetter, req.NamespacedName.String()); err != nil {
+		reason, err := validateDRPolicy(ctx, drpolicy, r.APIReader, r.ObjectStoreGetter, req.NamespacedName.String())
```
Comment (ShyamsundarR) [inline controllers/drpolicy_controller.go:80]: (nit) `err` gets defined here for the main function body, rest of the function has `err` in it's own sub-blocks, would like it if the same was retained here as well. (possibly `reason` in the future needs to be outside the block for metroDR and hence both get defined here)

**Review by ShyamsundarR — APPROVED** [review]

File: controllers/drplacementcontrol.go [hunk 2]
```diff
@@ -880,6 +881,22 @@ func (d *DRPCInstance) createVRGManifestWork(homeCluster string) error {
 	return nil
 }
 
+func (d *DRPCInstance) generateVRG() rmn.VolumeReplicationGroup {
+	vrg := rmn.VolumeReplicationGroup{
```
Comment (BenamarMk) [inline controllers/drplacementcontrol.go:885]: (nit) Should a vrg pointer be used instead?

**Review by BenamarMk — APPROVED** [review]

File: controllers/drpolicy_controller.go (hunk 1 above)
Comment (raghavendra-talur) [reply controllers/drpolicy_controller.go:80]: Yes, the reason is coming in from one of the called functions and we would like to keep it that way. I think we should refactor the other two calls also at some point in the future.

**Comment by raghavendra-talur** [comment]
@ShyamsundarR This PR is now rebased on top of #346 and #358. Can be merged with automatic rebase if those two are merged as-is.

**Review by ShyamsundarR — APPROVED** [review]

**Pull request merged**